Automattic/pocket-casts-ios
# App Clip: Add target and player implementation

| 📘 Part of: #2549 |
|:---:|

Implements #2550

* Adds the App Clip target
* Adds necessary files and ifdefs for Now Playing functionality
* Adds SKOverlay for upgrade prompting

| | | |
|-|-|-|
| ![Simulator Screenshot - iPhone 16 - 2024-12-17 at 21 13 23](https://github.com/user-attachments/assets/c6ec386e-34bb-4899-bec5-c6c7ba496e5c) | ![Simulator Screenshot - iPhone 16 - 2024-12-17 at 21 13 18](https://github.com/user-attachments/assets/a189ce5b-7668-4da6-97cb-2e8b89a5cdca) | ![Simulator Screenshot - iPhone 16 - 2024-12-17 at 21 16 32](https://github.com/user-attachments/assets/ff2cfabc-54eb-4b46-9aa0-3f36462d089a) |

## Review

This is a large PR and I apologize for not being able to break it up smaller. The VAST majority of the changes are in the `project.pbxproj` file to add various files to the App Clip target and many more are from the Asset Catalog JSON files.

If you filter to only the `swift` files, using the "File filter" in Github's UI it should be easier to parse through. All of the modified existing swift files are `ifdefs` for App Clip support. The new App Clip files are just for adding the Now Playing screen, handling the user activity, and showing the `SKOverlay`.

## To test

* Build and run the "Pocket Casts App Clip" target
* Ensure the Hidden Brain episode loads and plays
* Test playback, sleep timer, and air play functionality
* Ensure that the SKOverlay upgrade banner appears after 10 seconds
* Leave the app and check the App Library at the end of your Home Screens. You should see a "Webclips" group which includes the Pocket Casts logo with dashes representing the App Clip. Tapping it will bring you back into the App Clip.

## Checklist
